lib/dal/src/schema/variant.rs

```diff
@@ -968,22 +1072,431 @@ impl SchemaVariant {
+            )
+            .await?
+        {
+            if let (EdgeWeightKind::Func(kind), Some(node_weight)) = (
+                edge_weight.kind(),
+                workspace_snapshot.get_node_weight(target_idx).await.ok(),
+            ) {
+                if kind.to_owned() == FuncKind::Qualification {
+                    code_gen_funcs.push(node_weight.id().into())
+                }
+            }
+        }
+        Ok(code_gen_funcs)
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct SchemaVariantMetadataJson {
+    /// Name for this variant. Actually, this is the name for this [`Schema`](crate::Schema), we're
+    /// punting on the issue of multiple variants for the moment.
+    pub name: String,
+    /// Override for the UI name for this schema
+    #[serde(alias = "menu_name")]
+    pub menu_name: Option<String>,
+    /// The category this schema variant belongs to
+    pub category: String,
+    /// The color for the component on the component diagram as a hex string
+    pub color: String,
+    #[serde(alias = "component_type")]
+    pub component_type: ComponentType,
+    pub link: Option<String>,
+    pub description: Option<String>,
+}
+
+/// The json definition for a [`SchemaVariant`](crate::SchemaVariant)'s [`Prop`](crate::Prop) tree (and
+/// more in the future).
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct SchemaVariantJson {
+    /// The immediate child [`Props`](crate::Prop) underneath "/root/domain".
+    #[serde(default)]
+    pub props: Vec<PropDefinition>,
+    /// The immediate child [`Props`](crate::Prop) underneath "/root/secrets".
+    #[serde(default)]
+    pub secret_props: Vec<PropDefinition>,
+    /// The immediate child [`Props`](crate::Prop) underneath "/root/secretsDefinition".
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub secret_definition: Option<Vec<PropDefinition>>,
+    /// The immediate child [`Props`](crate::Prop) underneath "/root/resource_value".
+    #[serde(default)]
+    pub resource_props: Vec<PropDefinition>,
+    /// Identity relationships for [`Props`](crate::Prop) underneath "/root/si".
+    #[serde(default)]
+    pub si_prop_value_froms: Vec<SiPropValueFrom>,
+
+    /// The input [`Sockets`](crate::Socket) and created for the [`variant`](crate::SchemaVariant).
+    #[serde(default)]
+    pub input_sockets: Vec<SocketDefinition>,
+    /// The output [`Sockets`](crate::Socket) and created for the [`variant`](crate::SchemaVariant).
+    #[serde(default)]
+    pub output_sockets: Vec<SocketDefinition>,
+    /// A map of documentation links to reference. To reference links (values) specify the key via
+    /// the "doc_link_ref" field for a [`PropDefinition`].
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub doc_links: Option<HashMap<String, String>>,
+}
+
+impl SchemaVariantJson {
+    //     pub fn to_spec(
```

Comment:
Why are we keeping this?

Reply:
this is functionality that will come back!